relentless/potential/pair.py

```diff
@@ -589,9 +589,123 @@ class Depletion(PairPotential):
+
+            Parameters
+            ----------
+            param : str
+                The parameter with respect to which to take the derivative.
+                (Can only be 'sigma_i' or 'sigma_j').
+
+            Returns
+            -------
+            float
+                The calculated derivative value.
+
+            Raises
+            ------
+            ValueError
+                If the parameter argument is not 'sigma_i' or 'sigma_j'.
+
+            """
+            if param == 'sigma_i':
+                return 0.5
+            elif param == 'sigma_j':
+                return 0.5
+            else:
+                raise ValueError('Unknown parameter')
+
+    class HighBound(core.DependentVariable):
+        r"""Dependent variable for the high bound of the depletion pair potential.
+
+        .. math::
+
+            high = \frac{1}{2}(\sigma_i+\sigma_j)+\sigma_d
+
+        Parameters
+        ----------
+        sigma_i : int/float or :py:class:`Variable`
+            The `sigma_i` parameter of the high bound.
+        sigma_j : int/float or :py:class:`Variable`
+            The `sigma_j` parameter of the high bound.
+        sigma_d : int/float or :py:class:`Variable`
+            The `sigma_d` parameter of the high bound.
+
+        """
+        def __init__(self, sigma_i, sigma_j, sigma_d):
+            super().__init__(sigma_i=sigma_i, sigma_j=sigma_j, sigma_d=sigma_d)
+
+        @property
+        def value(self):
+            return 0.5*(self.sigma_i.value + self.sigma_j.value) + self.sigma_d.value
+
+        def _derivative(self, param):
+            """Calculates the derivative of the HighBound object with respect to its parameters.
+
+            Parameters
+            ----------
+            param : str
+                The parameter with respect to which to take the derivative.
+                (Can only be 'sigma_i', 'sigma_j', or 'sigma_d').
+
+            Returns
+            -------
+            float
+                The calculated derivative value.
+
+            Raises
+            ------
+            ValueError
+                If the parameter argument is not 'sigma_i', 'sigma_j', or 'sigma_d'.
+
+            """
+            if param == 'sigma_i':
+                return 0.5
+            elif param == 'sigma_j':
+                return 0.5
+            elif param == 'sigma_d':
+                return 1.0
+            else:
+                raise ValueError('Unknown parameter')
+
     def __init__(self, types, shift=False):
         super().__init__(types=types, params=('P','sigma_i','sigma_j','sigma_d'))
 
+    def energy(self, pair, r):
+        # Override parent method to set rmin and rmax as low and high bounds
+        if self.coeff[pair]['rmin'] is False:
+            self.coeff[pair]['rmin'] = self.LowBound(self.coeff[pair]['sigma_i'],
+                                                     self.coeff[pair]['sigma_j'])
```

Comment:
Let's not set `rmin` by default. The user can always construct this parameter themselves.

(Also apply to the `force` and `derivative` methods.)